File: frontend/packages/main/src/parts/sample-source-part.ts
import { KBObject, KBObjectPart } from "@genexusm-sdk/architecture-common";
import { Consts } from "../consts";

export class SampleSourcePart extends KBObjectPart {

    constructor(kbObject: KBObject) {
        super(kbObject, Consts.PartClasses.SAMPLE_SOURCE_PART);
    }
}

File: backend/GXServicesSampleExtension/GXServicesSampleExtension/Parts/SampleSourcePart.cs
using System.Runtime.InteropServices;
using System.Text;
using Artech.Architecture.Common.Descriptors;
using Artech.Architecture.Common.Objects;
using Artech.Genexus.Common.Parts;

namespace GXServicesSampleExtension.Parts;

[Guid("D74BEEDF-76C7-4617-8EC3-EFBE0A9572BB")]
[KBObjectPartDescriptor("SampleSourcePart", HasInterface = true)]
public class SampleSourcePart : KBObjectPart, ISource
{
    public SampleSourcePart(KBObject kbObject) 
        : base(typeof(SampleSourcePart).GUID, kbObject)
    {
        Source = string.Empty;
    }

    public string Source { get; set; }

    protected override byte[] SerializeData()
    {
        return Encoding.UTF8.GetBytes(Source);
    }

    protected override void DeserializeData(byte[] data)
    {
        Source = Encoding.UTF8.GetString(data);
    }
}